portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessdeliveryapi/DocumentAPI.macro

```diff
@@ -29,6 +29,116 @@ definition {
+		}
+		else {
+			var api = "sites/${siteId}/documents/by-external-reference-code/${externalReferenceCode}";
+		}
+
+		var curl = '''
+			${portalURL}/o/headless-delivery/v1.0/${api} \
+			-u test@liferay.com:test \
+			-H accept: application/json
+		''';
+
+		var curl = JSONCurlUtil.delete("${curl}");
+
+		return "${curl}";
+	}
+
+	macro _filterDocumentInAssetLibrary {
+		Variables.assertDefined(parameterList = "${assetLibraryId},${filtervalue}");
+
+		var portalURL = JSONCompany.getPortalURL();
+
+		var curl = '''
+			${portalURL}/o/headless-delivery/v1.0/asset-libraries/${assetLibraryId}/documents?filter=${filtervalue} \
+			-u test@liferay.com:test \
+			-H accept: application/json
+		''';
+
+		var curl = JSONCurlUtil.get("${curl}");
+
+		return "${curl}";
+	}
+
+	macro _getDocumentsByErcInAssetLibrary {
+		Variables.assertDefined(parameterList = "${assetLibraryId},${externalReferenceCode}");
+
+		var portalURL = JSONCompany.getPortalURL();
+
+		var curl = '''
+			${portalURL}/o/headless-delivery/v1.0/asset-libraries/${assetLibraryId}/documents/by-external-reference-code/${externalReferenceCode} \
+			-u test@liferay.com:test \
+			-H Content-Type: application/json
+		''';
+
+		var curl = JSONCurlUtil.get("${curl}");
+
+		return "${curl}";
+	}
+
+	macro _updateDocumentByErc {
+		Variables.assertDefined(parameterList = "${externalReferenceCode}");
+
+		if (!(isSet(filePath) || isSet(title) || isSet(updateExternalReferenceCode))) {
+			TestUtils.fail(message = "Please set at least one property to update the document.");
+		}
+
+		if (isSet(assetLibraryId)) {
+			var api = "asset-libraries/${assetLibraryId}/documents/by-external-reference-code/${externalReferenceCode}";
+		}
+		else {
+			var api = "asset-libraries/${siteId}/documents/by-external-reference-code/${externalReferenceCode}";
+		}
+
+		var portalURL = JSONCompany.getPortalURL();
+
+		var curl = '''
+			${portalURL}/o/headless-delivery/v1.0/${api} \
+			-u test@liferay.com:test \
+			''';
+		var delimiter = " ";
+
+		if (isSet(filePath)) {
+			var fileBody = "-F file=@&quot;${filePath}&quot;";
+
+			var curl = StringUtil.add("${curl}", "${fileBody}", "${delimiter}");
+		}
+
+		if (isSet(title) && isSet(updateExternalReferenceCode)) {
+			var documentBody = '''
+				-F document={"externalReferenceCode": "${updateExternalReferenceCode}", "title": "${title}"}
+			''';
+
+			var curl = StringUtil.add("${curl}", "${documentBody}", "${delimiter}");
+		}
+		else {
+			if (isSet(title)) {
+				var documentBody = '''
+					-F document={"title": "${title}"}
+				 ''';
+			}
+			else if (isSet(updateExternalReferenceCode)) {
+				var documentBody = '''
+					-F document={"externalReferenceCode": "${updateExternalReferenceCode}"}
+				''';
+			}
+
+			var curl = StringUtil.add("${curl}", "${documentBody}", "${delimiter}");
+		}
+
+		var curl = JSONCurlUtil.put("${curl}");
+
+		return "${curl}";
```

Comment:
I was thinking we could work here on a level of the body before creating a curl, this way maybe we would only need two 'if' statements we could combine them freely, but I don't have a lot of time to create a proposal, so this solution is fine by me 😅